src/plugins/intel_gpu/src/kernel_selector/kernels/activation/activation_kernel_base.cpp

```diff
@@ -22,8 +22,10 @@ ActivationKernelBase::DispatchData ActivationKernelBase::SetDefault(const activa
                                                                          {Tensor::DataChannelName::Y}};
         dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, arg.engineInfo, in_layout, out_layout, dims_by_gws);
     } else if (out_layout == DataLayout::b_fs_yx_fsv16 || out_layout == DataLayout::b_fs_yx_fsv32) {
-        dispatchData.gws = {Align(out.Feature().v, 16) * out.Batch().v, out.X().v, out.Y().v};
-        dispatchData.lws = {16, 1, 1};
+        const size_t feature_batch = out.Feature().v * out.Batch().v;
+        const size_t lws0 = std::min<size_t>(16, feature_batch);
+        dispatchData.lws = {lws0, 1, 1};
+        dispatchData.gws = {Align(feature_batch, lws0), out.X().v, out.Y().v};
```

Comment:
I think this is only for the case when batch * feature < 16. 
the original problem will be same when e.g., [17, 1, 384]..

Reply:
For the case `[17, 1, 384]`, `gws` will be `{32, 1, 384}` and I also modified (after your review) `activation_ref.cl` to calculate only valid range.